portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -104,7 +104,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     UserService userService = UserServiceFactory.getUserService();
     
     // Get the user, comment, and image from the form and add it to the array.
-    String username = userService.getCurrentUser().getEmail();
+    String username = getUserNickname(userService.getCurrentUser().getUserId());
```

Comment:
Is this supposed to be a username or a nickname? It's not completely clear because this line references 'username' and 'getNickname'. I would try to make these more consistent:

```suggestion
    String nickname = getUserNickname(userService.getCurrentUser().getUserId());
    
    OR
    
    String username = getUsername(userService.getCurrentUser().getUserId());
```

Reply:
Done! I'll stick with nickname.